food-search-app/src/main/java/com/example/foodsearchapp/controller/AuthController.java

```diff
@@ -0,0 +1,118 @@
+
+    @Autowired
+    PasswordEncoder passwordEncoder;
+
+    @Autowired
+    Utility utility;
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authUser(@RequestBody LoginRequest loginRequest) {
+
+        Authentication auth = authenticationManager
+                .authenticate(new UsernamePasswordAuthenticationToken(
+                        loginRequest.getUsername(),
+                        loginRequest.getPass()));
+
+        SecurityContextHolder.getContext().setAuthentication(auth);
+        String jsonWebToken = utility.generateJsonWebToken(auth);
+
+        UserDetailsImpl userDetails = (UserDetailsImpl) auth.getPrincipal();
+        List<String> statuses = userDetails.getAuthorities().stream()
+                .map(GrantedAuthority::getAuthority)
+                .collect(Collectors.toList());
+
+        return ResponseEntity.ok(new JsonWebTokenResponse(jsonWebToken,
+                userDetails.getId(),
+                userDetails.getUsername(),
+                userDetails.getEmail(),
+                statuses));
+    }
+
+    @PostMapping("/registry")
+    public ResponseEntity<?> registerUser(@RequestBody RegistrationRequest registrationRequest) {
+
+        if (userRepo.existsByUsername(registrationRequest.getUsername())) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(new MessResponse("Error: Username exist"));
+        }
+
+        if (userRepo.existsByEmail(registrationRequest.getEmail())) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(new MessResponse("Error: Email exist"));
+        }
+
+        User user = new User(registrationRequest.getUsername(),
+                registrationRequest.getEmail(),
+                passwordEncoder.encode(registrationRequest.getPass()));
+
+        Set<String> requestStatus = registrationRequest.getStatuses();
+        Set<Status> status = new HashSet<>();
+
+        if (requestStatus == null) {
+            Status userStatus = statusRepo
+                    .findByName(EStatus.STATUS_USER)
+                    .orElseThrow(() -> new RuntimeException("Error, USER status not found"));
+            status.add(userStatus);
+        } else {
+            requestStatus.forEach(r -> {
+                if ("restaurant".equals(r)) {
+                    Status restaurantStatus = statusRepo
+                            .findByName(EStatus.STATUS_RESTAURANT)
+                            .orElseThrow(() -> new RuntimeException("Error, RESTAURANT status not found"));
+                    status.add(restaurantStatus);
+                } else {
+                    Status userStatus = statusRepo
+                            .findByName(EStatus.STATUS_USER)
+                            .orElseThrow(() -> new RuntimeException("Error, USER status not found"));
+                    status.add(userStatus);
+                }
+            });
+        }
+        user.setStatuses(status);
```

Comment:
it's business logic. Service layer responded for it. Controller should only receive request and return response. In mvc make redirections. That's it. You should create service.